lib/modules/hosts/giphy.js

```diff
@@ -1,27 +1,35 @@
+import { ajax } from '../../environment';
+import logo from '../../images/hosts/giphy-logo.png';
+
 export default {
 	moduleID: 'giphy',
 	name: 'giphy',
 	domains: ['giphy.com'],
-	logo: '//giphy.com/static/img/favicon.png',
-	detect: ({ pathname }) => (/^\/gifs\/(?:\w+-)*(.*?)(\/html5)?$/i).exec(pathname),
-	handleLink(href, [, id, isHtml5]) {
+	logo,
+	detect: ({ pathname }) => (/^(?:\/gifs|\/media|)\/(?:\w+-)*([^/.]*)(\/html5)?.*$/i).exec(pathname),
```

Comment:
Also, we can remove `(\/html5)?`, see below (my other suggestions still apply as-is)